exporters/prometheus/src/exporter_utils.cc

```diff
@@ -167,6 +167,156 @@ std::string PrometheusExporterUtils::SanitizeNames(std::string name)
+                                                  {"By", "bytes"},
+                                                  {"KiBy", "kibibytes"},
+                                                  {"MiBy", "mebibytes"},
+                                                  {"GiBy", "gibibytes"},
+                                                  {"TiBy", "tibibytes"},
+                                                  {"KBy", "kilobytes"},
+                                                  {"MBy", "megabytes"},
+                                                  {"GBy", "gigabytes"},
+                                                  {"TBy", "terabytes"},
+                                                  {"B", "bytes"},
+                                                  {"KB", "kilobytes"},
+                                                  {"MB", "megabytes"},
+                                                  {"GB", "gigabytes"},
+                                                  {"TB", "terabytes"},
+                                                  // SI
+                                                  {"m", "meters"},
+                                                  {"V", "volts"},
+                                                  {"A", "amperes"},
+                                                  {"J", "joules"},
+                                                  {"W", "watts"},
+                                                  {"g", "grams"},
+                                                  // Misc
+                                                  {"Cel", "celsius"},
+                                                  {"Hz", "hertz"},
+                                                  {"1", ""},
+                                                  {"%", "percent"},
+                                                  {"$", "dollars"}};
+  auto res_it = units.find(unit_abbreviation);
+  if (res_it == units.end())
+  {
+    return unit_abbreviation;
+  }
+  return res_it->second;
+}
+
+std::string PrometheusExporterUtils::GetPrometheusPerUnit(const std::string &per_unit_abbreviation)
+{
+  static std::map<std::string, std::string> per_units{
+      {"s", "second"}, {"m", "minute"}, {"h", "hour"}, {"d", "day"},
+      {"w", "week"},   {"mo", "month"}, {"y", "year"}};
+  auto res_it = per_units.find(per_unit_abbreviation);
+  if (res_it == per_units.end())
+  {
+    return per_unit_abbreviation;
+  }
+  return res_it->second;
+}
+
+std::string PrometheusExporterUtils::RemoveUnitPortionInBraces(const std::string &unit)
+{
+  return std::regex_replace(unit, CHARACTERS_BETWEEN_BRACES_PATTERN, "");
+}
+
+std::string PrometheusExporterUtils::ConvertRateExpressedToPrometheusUnit(
+    const std::string &rate_expressed_unit)
+{
+  if (rate_expressed_unit.find("/") == std::string::npos)
+  {
+    return rate_expressed_unit;
+  }
+
+  std::vector<std::string> rate_entities;
+  size_t pos = rate_expressed_unit.find("/");
```

Comment:
nit

do not execute the find twice, save the result in `pos` and test for `pos == std::string::npos` at the start of this method.